File: algorithms/01051_height-checker/cpp/Solution1.cpp
#include<iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
  int heightChecker(vector<int> &heights) {
    int count = 0;
    vector<int> temp = heights;
    sort(heights.begin(), heights.end());
    for (int i = 0; i < heights.size(); ++i) {
      if (temp[i] != heights[i]) {
        count++;
      }
    }
    return count;
  }
};

int main() {
  vector<int> nums = {1, 1, 4, 6, 1, 3};
  int a = Solution().heightChecker(nums);
  cout << a << " ";
  return 0;
}

File: leetcode/algorithms/189. 旋转数组/java/Solution1.java
// 这个会超时
class Solution1 {
    public static void rotate(int[] nums, int k) {
      int prev;
      int temp;
      for (int i = 0; i < k; i++) {
        prev = nums[nums.length-1];
        for (int j = 0; j < nums.length; j++) {
          temp = nums[j];
          nums[j] = prev;
          prev = temp;
        }
      }
    }
  }